File: src/net/reichholf/dreamdroid/helpers/enigma2/requesthandler/impl/VolumeRequestHandler.java
/* © 2010 Stephan Reichholf <stephan at reichholf dot net>
 * 
 * Licensed under the Create-Commons Attribution-Noncommercial-Share Alike 3.0 Unported
 * http://creativecommons.org/licenses/by-nc-sa/3.0/
 */

package net.reichholf.dreamdroid.helpers.enigma2.requesthandler.impl;

import java.util.ArrayList;

import net.reichholf.dreamdroid.helpers.ExtendedHashMap;
import net.reichholf.dreamdroid.helpers.SimpleHttpClient;
import net.reichholf.dreamdroid.helpers.enigma2.URIStore;
import net.reichholf.dreamdroid.helpers.enigma2.Volume;

import org.apache.http.NameValuePair;

/**
 * @author sre
 *
 */
public class VolumeRequestHandler {
	public String get(SimpleHttpClient shc, ArrayList<NameValuePair> params){
		if (shc.fetchPageContent(URIStore.VOLUME, params)) {
			return shc.getPageContentString();
		}

		return null;
	}
	
	public ExtendedHashMap parse(String xml){
		return Volume.parse(xml);
	}
}

File: app/src/net/reichholf/dreamdroid/helpers/enigma2/requesthandler/VolumeRequestHandler.kt
/* © 2010 Stephan Reichholf <stephan at reichholf dot net>
 *
 * Licensed under the Create-Commons Attribution-Noncommercial-Share Alike 3.0 Unported
 * http://creativecommons.org/licenses/by-nc-sa/3.0/
 */

package net.reichholf.dreamdroid.helpers.enigma2.requesthandler

import net.reichholf.dreamdroid.enigma.Volume
import net.reichholf.dreamdroid.enigma.VolumeParser
import net.reichholf.dreamdroid.helpers.enigma2.URIStore

/**
 * @author sre
 */
class VolumeRequestHandler : AbstractSimpleRequestHandler(URIStore.VOLUME) {
    fun parseVolume(xml: String?): Volume {
        if (xml == null) {
            return Volume()
        }
        return VolumeParser.parse(xml) ?: Volume()
    }
}
